events/class_models/possible_participant_model.py

```diff
@@ -0,0 +1,56 @@
+from django.db import models
+from django.core.exceptions import ValidationError
+from .meeting_models import PossibleMeeting
+from .participant_model import EventParticipant
+from .event_models import time_format
+
+
+class PossibleParticipantsQuerySet(models.QuerySet):
+
+    def get_all_date_participants(self, meeting_id):
+        return self.filter(possible_meeting_id=meeting_id)
+
+    def get_all_possible_participants(self, event_id):
+        return self.filter(possible_meeting_id__participant_id__event_id=event_id)
+
+    def remove_all_possible_meeting_participants(self, meeting_id):
+        return self.get_all_date_participants(meeting_id).delete()
+
+    def remove_all_event_participants(self, event_id):
+        return self.get_all_possible_participants(event_id).delete()
+
+
+class PossibleParticipant(models.Model):
+    participant_id = models.ForeignKey(EventParticipant, on_delete=models.CASCADE)
+    possible_meeting_id = models.ForeignKey(PossibleMeeting, on_delete=models.CASCADE)
+
+    objects = PossibleParticipantsQuerySet.as_manager()
+
+    def clean(self) -> None:
+        validate_unique_possible_participant(self.participant_id, self.possible_meeting_id)
+        validate_event_participant(self.participant_id.event_id, self.possible_meeting_id.participant_id.event_id)
+        return super().clean()
+
+    def save(self, *args, **kwargs):
+        self.clean()
+        return super().save(*args, **kwargs)
+
+    def __str__(self) -> str:
+        return (f"participant: {self.participant_id.user_id.email}"
+                f" event: {self.possible_meeting_id.participant_id.event_id.title}"
+                f" {time_format(self.possible_meeting_id.date_time_start)} -"
+                f" {time_format(self.possible_meeting_id.date_time_end)} ")
+
+
+# validate if the current participant is participanting in the current event
+def validate_event_participant(participant_event, event):
+    if participant_event != event:
+        raise ValidationError("The participant is not part of this event")
```

Comment:
```suggestion
def validate_event_participant(participant_event, event):
    """ validate if the current participant is participanting in the current event """
    if participant_event != event:
        raise ValidationError("The participant is not part of this event")
```

Reply:
Why write the comment inside the function? 
And if the comment is only one line so why to write it with """ and not # ?

Reply:
because this comment is basically the definition of the function,
and in python when we write a definition of a function we would write it in this way.

If this is a comment on the function in my opinion it should be inside the function.

Reply:
I prefer to write functions' comments above the function (for me it's more readable), 
but if it's more readable to write the comments inside the function and that is the convention, so I will change it 😄

Reply:
thanks, 👍